src/main/scala/cognite/spark/v1/RawTableRelation.scala

```diff
@@ -102,13 +130,19 @@ class RawTableRelation(
 
     @transient lazy val rowConverter = getRowConverter(schema)
 
-    val partitionCursors =
-      CdpConnector
-        .clientFromConfig(config)
-        .rawRows(database, table)
-        .getPartitionCursors(filter, configWithLimit.partitions)
-        .unsafeRunSync()
-        .toVector
+    val streams: (GenericClient[IO], Option[Int], Int) => Seq[Stream[IO, RawRow]] = keysFilter match {
+      case Nil => {
+        val partitionCursors =
+          CdpConnector
+            .clientFromConfig(config)
+            .rawRows(database, table)
+            .getPartitionCursors(filter, configWithLimit.partitions)
+            .unsafeRunSync()
+            .toVector
+        getStreams(filter, partitionCursors)
+      }
+      case keys => getStreamsByKeys(keys.toArray)
```

Comment:
It should be safe to assume that when filtering on keys, the number of keys will be quite limited, so we can just retrieve them all in parallel (or possibly in parallel for smaller groups, but I think our sttp backend limits parallelism to something reasonable anyway).

We can even enforce this, throwing an error if the user requests more than 100 keys or something. Note that if a user does something like `select * from my.table where keys in (select key from other.table)`, it doesn't get pushed down by Spark anyway - keys must be specified explicitly, which is why it should be feasible to assume that the number of keys will be small, and to throw an error if not.

Reply:
I think throwing an error might be too blocking, and I've seen at least one query with over 100 hardcoded cases (i don't remember if it was a raw query)... maybe just ignoring the `key` filters and going back to the current behavior (with last updated time and columns filters only) would be better if too many keys are provided.